### ISSUE TITLE
simplify mesg-js

### ISSUE THREAD

**Issue by ilgooz** [mesg-foundation/mesg-js#35]
- [ ] Try extending EventEmitter in Stream
- [ ] Update types https://github.com/mesg-foundation/mesg-js/blob/master/src/client/service.ts
- [ ] RM public apis of application except gRPC APIs

**Comment by NicolasMahe** [comment]
- [ ] update proto and typescript definition when release core v0.6 is merged

**Comment by NicolasMahe** [comment]
- [ ] check implementation of `deployService(request: DeployServiceRequest, cb: (err: Error, data: DeployServiceReply) => void)`

**Comment by NicolasMahe** [comment]
Useful ressources:
https://github.com/carlessistare/grpc-promise
https://github.com/grpc-ecosystem/awesome-grpc#other-nodejs

```
 grpc_promise.promisifyAll(client);
```

**Comment by antho1404** [comment]
I don't think we need to force to use promises, let's keep it simple if people want to use promise the can promisify (which is a standard package in node).

Also it's confusing to use promises when you also have streams.

I don't really understand the "Try extending EventEmitter in Stream". Streams are using EventEmitters so I don't see the reason to use stream and add some development when we already have the event emitter.

**Comment by antho1404** [comment]
Also i would recommend to be as close as possible to the ts-protoc-gen. You can have a look in this PR https://github.com/mesg-foundation/mesg-js/pull/28

I think if the library can automatically be updated only based on the ts-protoc-gen that's perfect

**Comment by ilgooz** [comment]
Since we're going to publish a new major release, breaking changes aren't important. So, I think we should start using a generator to create the gRPC client and types. I'm freezing the development for now. I think we should continue on the PR mesg-foundation/mesg-js#28. 

Could you take look at to my new comment on the forum about websocket proxy [here](https://forum.mesg.com/t/make-a-websocket-server-to-expose-the-apis/150/2?u=ilgooz).

**Comment by antho1404** [comment]
Agree, we could remove all the application when then logic and just keep the generated files.

**Comment by NicolasMahe** [comment]
I think we should always keep the gRPC node version.
The Web version is not fully compatible with all gRPC functionalities and
is not yet officially maintain.

What about splitting the library in mesg-node and mesg-js or mesg-web (like
gRPC).
